File: leetcode/34/34.go
package main

import (
	"sort"
)

func searchRange(nums []int, target int) []int {
	begin := -1
	end := -1

	if len(nums) > 0 {
		begin = sort.SearchInts(nums, target)
		if begin < len(nums) && nums[begin] == target {
			end = sort.SearchInts(nums, target+1) - 1
		} else {
			begin = -1
		}
	}

	return []int{begin, end}
}

File: leetcode/34/34.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    vector<int> searchRange(vector<int> &nums, int target) {
        vector<int> res;
        int         begin = -1, end = -1;
        int         size = nums.size();
        if (size > 0) {
            auto it = lower_bound(nums.begin(), nums.end(), target);
            begin = it - nums.begin();
            if (it != nums.end() && *it == target) {
                it = lower_bound(nums.begin(), nums.end(), target + 1) - 1;
                end = it - nums.begin();
            } else
                begin = -1;
        }
        res.push_back(begin);
        res.push_back(end);
        return res;
    }
};